contentcuration/contentcuration/assets/channelList/views/Channel/ChannelModal.vue

```diff
@@ -0,0 +1,705 @@
+            maxlength="400"
+            rows="4"
+            counter
+          />
+          <VCardText v-if="channel.created">
+            {{ $tr('created', { date: new Date(channel.created) }) }}
+          </VCardText>
+          <VCardText v-if="channel.last_published">
+            {{ $tr('published', { date: new Date(channel.last_published) }) }}
+          </VCardText>
+          <template v-if="channel.edit">
+            <p>{{ $tr('deletePrompt') }}</p>
+            <VBtn class="upper" color="error" @click="deleteChannel">
+              {{ $tr('deleteChannel') }}
+            </VBtn>
+          </template>
+        </VFlex>
+        <VFlex xs1>
+          <ChannelStar
+            :channel="channel"
+          />
+        </VFlex>
+      </VLayout>
+      <VLayout v-if="channelDetails && channelDetails.resource_count" row wrap justify-center>
+        <VFlex xs4>
+          <VMenu offset-y>
+            <template v-slot:activator="{ on }">
+              <VBtn
+                color="primary"
+                dark
+                v-on="on"
+              >
+                {{ $tr('downloadReport') }}
+              </VBtn>
+            </template>
+            <VList>
+              <VListTile
+                v-for="(item, index) in downloadOptions"
+                :key="index"
+                :href="item.href"
+                download
+              >
+                <VListTileTitle>{{ item.title }}</VListTileTitle>
+              </VListTile>
+            </VList>
+          </VMenu>
+        </VFlex>
+      </VLayout>
+      <VLayout v-if="channelDetails && channelDetails.resource_count" row wrap>
+        <VFlex xs12 sm12 md8 justifyCenter>
+          <VTabs>
+            <VTab :key="0">
+              {{ $tr('whats_inside') }}
+            </VTab>
+            <VTabItem :key="0">
+              <VLayout row wrap>
+                <VFlex v-if="countBar" xs6>
+                  {{ $tr('resource_size') }}
+                  <span id="count_gauge" data-toggle="tooltip" data-placement="top" :title="$tr('resource_count', { count: channelDetails.resource_count}) + '-' + countBar.text">
+                    <b v-for="(fill, i) in countBar.filled" :key="i" class="count_icon" :class="fill ? 'filled' : ''">
+                      ▮
+                    </b>
+                  </span>
+                </VFlex>
+                <VFlex v-if="sizeBar" xs6>
+                  {{ $tr('storage') }}
+                  <span data-toggle="tooltip" data-placement="top" :title="channelDetails.resource_size + '-' + sizeBar.text">
+                    <VIcon v-for="(fill, i) in sizeBar.filled" :key="i">
+                      sd_storage
+                    </VIcon>
+                  </span>
+                </VFlex>
+              </VLayout>
+              <VLayout row wrap>
+                <template v-if="channelDetails.kind_count">
+                  <VFlex xs8>
+                    <!-- D3 Pie chart will be inserted here -->
```

Comment:
Could you also see if you can resolve the issue with asynchronous loading? I'm pretty sure that it's because d3 is looking for a selector, but you can probably use a more data-driven approach with vue-google-charts

 Here's what's happening here:
1. I opened Edraak, which starts a `get_topic_details` request
2. I opened Proyecto Descartes, which started another `get_topic_details` request
3. Proyecto Descartes finished first, so it loaded the correct information (orange pie chart)
4. Edraak finished and replaced the details (blue/green pie chart), but the active channel is still Proyecto Descartes

![details](https://user-images.githubusercontent.com/7447496/61414847-e2388000-a8a3-11e9-8d4c-bb85cde4b928.gif)

This is a little difficult to see here since the channels are fairly small, but it's a lot more noticeable with larger channels

Reply:
Yeah - the way I have this implemented currently would probably still break, so we should do some sanity checking here. The other alternative would be to just store the details in a channelDetailsMap object, and then just reference that (that way there wouldn't be any overlap).

Reply:
Current implementation now avoids this issue.

Reply:
Have updated the implementation to prevent this issue.